### PR TITLE
Stub cellOsk

### PR DESCRIPTION
**Pull request by al0xf**
A first and simple implementation of cellOsk (on screen keyboard) that for now just returns the word "rpcs3" as OSK input. This has been tested and is working in Terraria, Nier, and some homebrew. Specifically Terraria can now proceed in game. 

The next step would be to hook it up to the GUI so the user can enter any text. There may also be games that do not like the input "rpcs3" although for them this pull won't practically change anything.

As for creating a OSK GUI I was thinking about either adding a few lines of code in MsgDialog.cpp or adding a new .cpp, although in this case there would be a lot of pretty much duplicated code compared to MsgDialog.cpp. Feel free to suggest which way I should proceed. 

~~Edit: Also Travis is broken~~
> ~~Error: No formulae found in taps.~~
> ~~The command "if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo pip install cpp-coveralls requests[security]; else brew update; brew update; brew install glew wxwidgets llvm36; fi;" failed and exited with 1 during .~~
> ~~Your build has been stopped.~~

### CODE REVIEW

**Comment by MarioSonic2987** [comment]
> As for creating a OSK GUI I was thinking about either adding a few lines of code in MsgDialog.cpp or adding a new .cpp, although in this case there would be a lot of pretty much duplicated code compared to MsgDialog.cpp. Feel free to suggest which way I should proceed.

You can use mine too, which are in my own repository.

**Comment by AniLeo** [comment]
I believe Travis is stuck because it's attempting to install llvm36 and not finding the package (No available formula with the name "llvm36").
I've submitted a PR to try to fix it.

File: rpcs3/Emu/Cell/Modules/cellOskDialog.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	be_t<u32> allowOskPanelFlg;
+	be_t<u32> firstViewPanel;
+	CellOskDialogPoint controlPoint;
+	be_t<s32> prohibitFlgs;
+};
+
+//Actual input data
+struct CellOskDialogCallbackReturnParam
+{
+	CellOskDialogInputFieldResult result;
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellOskDialog.h]: Enum has endianness as any int type

File: rpcs3/Emu/Cell/Modules/cellOskDialog.h [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+{
+	be_t<u32> led;
+	be_t<u32> mkey;
+	be_t<u16> keycode;
+};
+
+struct CellOskDialogImeDictionaryInfo
+{
+	be_t<u32> targetLanguage;
+	const vm::bptr<char> dictionaryPath;
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellOskDialog.h]: Const usually applies to char, vm::bcptr for example

File: rpcs3/Emu/Cell/Modules/cellOskDialog.h [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+struct CellOskDialogLayoutInfo
+{
+	be_t<s32> layoutMode;
+	CellOskDialogPoint position;
+};
+
+struct CellOskDialogSeparateWindowOption
+{
+	CellOskDialogContinuousMode continuousMode;
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellOskDialog.h]: Enum

File: rpcs3/Emu/Cell/Modules/cellOskDialog.cpp [hunk 4]
```diff
@@ -1,117 +1,153 @@
...
+	cellOskDialog.warning("cellOskDialogUnloadAsync(OutputInfo=*0x%x)", OutputInfo);
+	OutputInfo->result = CELL_OSKDIALOG_INPUT_FIELD_RESULT_OK;
+
+	//TODO: Return actual input. 
+	*OutputInfo->pResultString = 'r'; 
+	*(OutputInfo->pResultString + 1) = 'p';
+	*(OutputInfo->pResultString + 2) = 'c';
+	*(OutputInfo->pResultString + 3) = 's';
+	*(OutputInfo->pResultString + 4) = '3';
+	*(OutputInfo->pResultString + 5) = 0x0; //Excplicit null termination.
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellOskDialog.cpp]: Consider using std::memcpy here

**Comment by al0xf** [comment]
Thank you, I have fixed those issues. Travis and Appveyor are doing their things now, sorry about the double commits, a line of code accidentally got lost in the first one.

**Pull request merged**